### DISCUSSION TITLE
New Component: Alert

### DISCUSSION THREAD

**Discussion by Queen-esther01** [heroui-inc/heroui#313] [Feature requests]
I would like to suggest the alert component. Similar to toast, however,
it can be used to share larger chunks of information in page, can also be used as an
alternative to toasts. It displays by default and does not leave until the user removes (if needed)
e.g info alert, warning alert, success alert, etc

I have considered using the modals as an alternative solution to a page alert
but it doesn't quite seem to do the trick because modals need to be triggered. If 
triggered by default, users can close this without actually reading the message. It seems
like a good idea to have an alert that stays on the page reminding users what to do.

**Additional context**
![bootstrap-alert-messages-with-icons](https://user-images.githubusercontent.com/35095984/154036499-9c066196-0b8a-45fe-bc73-7d5c58dff675.png)

**Comment by okinawaa** [comment]
some alert of toast component are needed !

**Reply by power-f-GOD** [reply]
Any updates on this?

**Comment by cruzluna** [comment]
Definitely need an Alert component. And it would be especially useful it it had a promise API feature similar to [React hot toast](https://github.com/timolins/react-hot-toast).

**Comment by shahzeb1** [comment]
Bumping this, I'm also happy to help contribute to this one.

**Comment by nayanrabiul** [comment]
I was surprised to see no alert component. How did they miss that, especially considering it is a pretty good React UI library?

**Comment by Corepex** [comment]
Looking forward to a alert or toast component

**Comment by andrewjunggg** [comment]
I was searching for the alert component this morning but couldn't find one. I'm looking forward to it!

**Comment by benmcmaster** [comment]
+1

**Comment by myanpetra99** [comment]
up

**Comment by alexlee85** [comment]
+1

**Comment by vitalijalbu** [comment]
any news about that??

**Comment by winchesHe** [comment]
https://nextuioss.featurebase.app/p/alert-4 Here the roadmap about alert

**Comment by shahriar0247** [comment]
when will it be done?

**Reply by winchesHe** [reply]
Next release, currently under development

**Reply by liqingitt** [reply]
> 下一版本目前正在开发中

When will it be released? This component is really needed.

**Comment by Deroswent** [comment]
any updates?

**Comment by Pszz** [comment]
**I think you can try to refer to them.**

https://sonner.emilkowal.ski/
https://www.npmjs.com/package/react-toastifyt

**Comment by soranoo** [comment]
looking forward to it!

**Comment by jessesulo** [comment]
+1

**Comment by WZhijun154** [comment]
+1

**Comment by vitalijalbu** [comment]
guys, just use **shadcn** or **radix-themes**. they are awesome... **forgot about nextui**... it's almost 1-2 years we want this damn Toast and Alert... Drawer included...

**Reply by jessesulo** [reply]
Some companies have been using Nextui for a bit we don’t wanna convert a ton of repos. It’s such a shame

**Reply by soranoo** [reply]
sadly shadcn and nextui can't work together in some cases, for example, a shadcn modal with nextui date picker inside and the date picker will fail to open.

**Comment by keiohtani** [comment]
+1

**Comment by wingkwong** [comment]
Just an update. This is planned to release in v2.5.0.

**Comment by wingkwong** [comment]
https://www.heroui.com/docs/components/alert
